enterprise/reporting/error_reporting.go

```diff
@@ -375,6 +382,27 @@ func (edr *ErrorDetailReporter) mainLoop(ctx context.Context, c types.SyncerConf
 			if err != nil {
 				edr.log.Errorf("[ Error Detail Reporting ]: Error deleting local reports from %s: %v", ErrorDetailReportsTable, err)
 			}
+			vacuumStart := time.Now()
+			var sizeEstimate int64
+			if err := dbHandle.QueryRowContext(
+				ctx,
+				fmt.Sprintf(`SELECT pg_table_size(oid) from pg_class where relname='%s';`, ErrorDetailReportsTable),
+			).Scan(&sizeEstimate); err != nil {
+				edr.log.Errorn(
+					`[ Error detail Reporting ]: Error getting table size estimate`,
+					logger.NewErrorField(err),
+				)
+			}
+			if sizeEstimate > config.GetInt64("Reporting.errorReporting.vacuumThresholdBytes", 5*bytesize.GB) {
+				vaccumStatement := fmt.Sprintf("vacuum full analyze %s;", ErrorDetailReportsTable)
```

Comment:
Safer approach to passing the table name:
```suggestion
				vaccumStatement := fmt.Sprintf("vacuum full analyze %s", pq.QuoteIdentifier (ErrorDetailReportsTable))
```